### ISSUE TITLE
[YouTube] RegexNotFoundError: Unable to extract JS player URL

### ISSUE THREAD

**Issue by BlackPropeller** [rg3/youtube-dl#4717]
Command: `./youtube-dl -s --verbose http://www.youtube.com/watch?v=LZ-XtBvPj0I`

Output:
```
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-s', '--verbose', 'http://www.youtube.com/watch?v=LZ-XtBvPj0I']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.01.15.1
[debug] Python version 2.6.6 - Linux-3.10.23-xxxx-std-ipv6-64-x86_64-with-centos-6.6-Final
[debug] exe versions: ffmpeg 0.6.5, ffprobe 0.6.5
[debug] Proxy map: {}
[youtube] LZ-XtBvPj0I: Downloading webpage
[youtube] LZ-XtBvPj0I: Downloading video info webpage
[youtube] LZ-XtBvPj0I: Extracting video information
ERROR: Unable to extract JS player URL; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "./youtube-link/youtube_dl/extractor/youtube.py", line 836, in _real_extract
    raise ValueError('Could not find ytplayer.config')  # caught below
ValueError: Could not find ytplayer.config
Traceback (most recent call last):
  File "./youtube-link/youtube_dl/YoutubeDL.py", line 600, in extract_info
    ie_result = ie.extract(url)
  File "./youtube-link/youtube_dl/extractor/common.py", line 260, in extract
    return self._real_extract(url)
  File "./youtube-link/youtube_dl/extractor/youtube.py", line 1022, in _real_extract
    embed_webpage if age_gate else video_webpage, 'JS player URL')
  File "./youtube-link/youtube_dl/extractor/common.py", line 516, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
RegexNotFoundError: Unable to extract JS player URL; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

Other affected URLs:
`http://www.youtube.com/watch?v=OVyB-ai9z3I`
`http://www.youtube.com/watch?v=zHQAElxHdHg`
`http://www.youtube.com/watch?v=x_OCUERNKLw`
`http://www.youtube.com/watch?v=lI7GjcpdI4A`

The url/video below says it's unavailable but for some reason it still shows the same 'JS Player URL' error: `http://www.youtube.com/watch?v=gc4lQvU9BCs`

**Comment by docsuleman** [comment]
issue is fixed in master. kindly update youtube_dl/extractor/youtube.py file

**Comment by phihag** [comment]
@docsuleman This bug report is for a release version; and we have not changed *anything* in master since the 2015.01.15.1 release.

**Comment by phihag** [comment]
Mmm, all mentioned URLs work fine for me. Can you post the (warning: large) output you get when you add `--dump-intermediate-pages` to the command line?

**Comment by BlackPropeller** [comment]
hmm that's strange. Here's the output from `--dump-intermediate-pages`:
http://pastebin.com/15p3xx5Z

**Comment by phihag** [comment]
According to that output, the video webpage is blocked in your country. Do you see the error message `The uploader has not made this video available in your country.` when visiting https://www.youtube.com/watch?v=LZ-XtBvPj0I from the same machine that youtube-dl is running on?

In any case, strangely enough, the metadata seems to be present, so we may just need to conjure a player URL to get the download running. However, I don't think I'll be able to do that without a way to test. Can you tell me the ISP/country where you are, or even better, give me access to a machine/proxy/VPN in that network? Feel free to [contact me in private](mailto:phihag@phihag.de).

**Comment by eerie** [comment]
Same here:
youtube-dl --verbose https://www.youtube.com/watch?v=ImHOSOML4Gkb 2>&1 | sprunge
http://sprunge.us/abGg

**Comment by dstftw** [comment]
@eerie update to 2015.01.15.1.

**Comment by eerie** [comment]
Sorry, I had it installed with pacman and the latest upstream release is not yet in the repos. Now, manually installed, it works.

**Comment by BlackPropeller** [comment]
@phihag okay, I'll setup a proxy and will email you the user/pass in a couple minutes. Thank you for investigating this!  oh and just an fyi, the server is in Montreal, Quebec, Canada.

**Comment by ad-m** [comment]
```
$ youtube-dl --verbose -s --extract-audio --audio-format mp3 "https://www.youtube.com/watch?v=EJUL_TrT_q8&list=RDEJUL_TrT_q8" --get-filename  
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['--verbose', '-s', '--extract-audio', '--audio-format', 'mp3', 'https://www.youtube.com/watch?v=EJUL_TrT_q8&list=RDEJUL_TrT_q8', '--get-filename']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.01.16
[debug] Python version 2.7.5+ - Linux-2.6.32-042stab102.9-x86_64-with-Ubuntu-13.10-saucy
[debug] exe versions: avconv 0.8.13-6, avprobe 0.8.13-6, ffprobe 0.8.13-6
[debug] Proxy map: {}
Ras Luta - Jak promień słońca-EJUL_TrT_q8.m4a
Ras Luta - Mała-V10QBVdIS8E.m4a
Mesajah - Każdego dnia-WbTuQ4S4jQc.m4a
Junior Stress - Znam Ten Stan-J2aRGzpin7o.m4a
Bas Tajpan - Trzy życzenia-42h5gOzj5DE.m4a
Eastwest Rockers - Kawałek Szczęścia [Official Video]-KhJBnwQE8Fo.m4a
Ras Luta - Milosc w kazdym domu-FGLAjbdyA0s.m4a
Ras Luta - Walcz caly czas-70h0m5MIu84.m4a
♫ Junior Stress - Kiedy nie myślę o niczym ♫-WXJDREmFJyQ.m4a
TaLLib - moja lady-GqAUB8jmpqg.m4a
EastWest Rockers - Dokąd tak biegniesz-hDNd1ZYb_IM.m4a
Eastwest Rockers - Będę na pewno-TBz8gdy2xaY.m4a
GrubSon - Na szczycie (official video)-6BnT_wkuMBA.m4a
EastWest Rockers - Wstaje Rano (Feat. Junior Stress)-I8lPXMLvdPE.m4a
Ras Luta - Miłość to wszystko-DS6L3qZyOig.m4a
Ras Luta- Nie mam Hajsu-HFD3JRb9EBE.m4a
MESAJAH -  Tylko raz dane-7yjy38XKDfs.m4a
Bas Tajpan- Złap mnie za rękę + tekst-5IeKuaIGT-g.m4a
Ras Luta Jeśli Słyszysz-lKix6ozySb0.m4a
East West Rockers - Wszystkie dziewczyny-YWTP8tn1trY.m4a
ERROR: Unable to extract JS player URL; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/extractor/youtube.py", line 836, in _real_extract
    raise ValueError('Could not find ytplayer.config')  # caught below
ValueError: Could not find ytplayer.config
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/YoutubeDL.py", line 600, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/extractor/common.py", line 260, in extract
    return self._real_extract(url)
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/extractor/youtube.py", line 1022, in _real_extract
    embed_webpage if age_gate else video_webpage, 'JS player URL')
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/extractor/common.py", line 516, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
RegexNotFoundError: Unable to extract JS player URL; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

```
From OVH box.

**Comment by BlackPropeller** [comment]
Hey @phihag, did you have some times yet to take a look at this issue using the proxy i emailed you?

**Comment by idanPickit** [comment]
I have the same problem a lot, I have the latest python and latest youtube_dl.
I don't think its related to blocked content cause I have this issue with videos from my own country..

**Comment by samiq** [comment]
I don't know if it has to do with the fact that google turn on html5 player by default today http://venturebeat.com/2015/01/27/youtube-ditches-flash-for-html5-video-by-default/ most of my video url's stopped working this afternoon and the server is in the USA

**Comment by phihag** [comment]
The message "The uploader has not made this video available in your country." is a red hering, I get it too but it still works fine.

**Comment by phihag** [comment]
Thank you for the report! I believe this issue to be fixed in youtube-dl 2015.01.30.2 and newer. See [our FAQ](https://yt-dl.org/update) if you need help updating.
